### ISSUE TITLE
Negation doesn't work properly with matchBase

### ISSUE THREAD

**Issue by jevancc** [micromatch/micromatch#119]
## What is happening (but shouldn't):
Negation pattern does not work properly with option `matchBase` in `micromatch@3.1.6`.

## Code
```
var mm = require('micromatch');   
mm.isMatch("./x/y.js", "*.js", { matchBase: true });    
// true    
mm.isMatch("./x/y.js", "!*.js", { matchBase: true });    
// true, but it should be false    
mm.isMatch("./x/y.js", "**/*.js", { matchBase: true });    
// true    
mm.isMatch("./x/y.js", "!**/*.js", { matchBase: true });    
// false    
```

**Comment by jonschlinkert** [comment]
@jevan0307 thanks for the issue. This was fixed, published and released as 3.1.7. Please reopen if you're still having issues, or create a new issue. Thanks!

**Comment by jonschlinkert** [comment]
Hmm, looks like the appveyor test is failing on windows for node.js 6 and lower. I'm not sure why, but oddly that version is incorrectly building the regular expression. It's perfectly valid regex, but still... a mystery. I'll figure it out.
